### PR TITLE
Do Not Merge -- zones are groups and have their own physics

### PR DESCRIPTION
**Pull request by sethalves**
- entities and avatars have a parentID which can be set to the entityID of a zone
- objects that are children of zones have positions and rotations relative to that zone
- each zone has its own physics engine

### CODE REVIEW

**Comment by sethalves** [comment]
This is pretty far from ready.  Hopefully this PR will let people look it over and point out problems.  My hope is to get it to the point that it doesn't break non-grouped stuff so that it can be merged and finished without being on a branch which is so divergent from master..

**Comment by ZappoMan** [comment]
Very exciting... but why limit this to zones? I've been hoping to add this as a general feature to all entities... to just let any entity have a parent entity...

**Comment by hifi-gustavo** [comment]
Build console output for commit: d8c282390dc2027bd48f19a4fac8968d5f4f88f6 available at: https://gist.github.com/9d224a6fc04659b77bcb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6002.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6002.dmg

**Comment by sethalves** [comment]
I suppose anything could be a parent, but it gets a little strange when I think about anything have its own physics simulation.  The simulation needs clear bounds.

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: d8c282390dc2027bd48f19a4fac8968d5f4f88f6 available at: https://gist.github.com/9d224a6fc04659b77bcb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6002.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6002.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f509ba65e6d48a6bffa5b572f9e3159e34f594c1 available at: https://gist.github.com/9d224a6fc04659b77bcb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6002.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6002.dmg

File: libraries/entities-renderer/src/RenderableZoneEntityItem.cpp [hunk 1]
```diff
@@ -18,28 +18,70 @@
...
+#include <ZoneTracker.h>
 
 // Sphere entities should fit inside a cube entity of the same size, so a sphere that has dimensions 1x1x1
 // is a half unit sphere.  However, the geometry cache renders a UNIT sphere, so we need to scale down.
 static const float SPHERE_ENTITY_SCALE = 0.5f;
 
 EntityItemPointer RenderableZoneEntityItem::factory(const EntityItemID& entityID, const EntityItemProperties& properties) {
-    return std::make_shared<RenderableZoneEntityItem>(entityID, properties);
+
+    auto zoneTracker = DependencyManager::get<ZoneTracker>();
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/RenderableZoneEntityItem.cpp]: hmmm - can you explain this? are you planning on sending edits to the entity server for all items in a zone when a zone is changed??? I definitely do not like the sound of that design.

We should implement a hierarchy in a manner that allows child object to go "unedited" when their parent is moved/changed.

Comment (sethalves) [inline libraries/entities-renderer/src/RenderableZoneEntityItem.cpp]: no, there should be no need to do that.  entities with a parent have relative-to-parent position and rotation, so they don't need to be updated when a parent is.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5e60c3f4df363083cdd9d3572ad363c3049cd8dd available at: https://gist.github.com/9d224a6fc04659b77bcb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6002.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6002.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1739da99ad432ebd6f3dd8724396c4de0209b5ac available at: https://gist.github.com/9d224a6fc04659b77bcb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6002.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6002.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 11a92b9bbea7f7fea75aeffd6bf7a8564af167af available at: https://gist.github.com/9d224a6fc04659b77bcb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6002.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6002.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 439a65e950690483dbd7ba32042000ce13b3c9a9 available at: https://gist.github.com/9d224a6fc04659b77bcb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6002.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6002.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6fdae2b815bf3f5267933aaf39ffbcfce8bd9b13 available at: https://gist.github.com/9d224a6fc04659b77bcb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6002.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6002.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: bb390c535716c44f0c3104bb5e04553459630e4d available at: https://gist.github.com/9d224a6fc04659b77bcb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6002.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6002.dmg

**Comment by thoys** [comment]
I wonder if it would also be a nice possibility to be able to pause the physics simulation per zone (set a different FPS (0 to pause)).

**Comment by ZappoMan** [comment]
Can we close this PR? It's almost 2 months old...

**Comment by sethalves** [comment]
I'm still referring to it.  I'll close it after the Parents branch merges.

**Comment by ZappoMan** [comment]
so - I think the "parents branch" has been merged... you still need this PR?

**Pull request closed without merging**